lib/hotel_organizer.rb

```diff
@@ -0,0 +1,71 @@
+module Hotel
+  class HotelOrganizer
+    attr_reader :reservations, :rooms
+    
+    def initialize
+      @reservations = []
+      @rooms = []
+      
+      20.times.each do |i|
+        @rooms << Hotel::Room.new(i + 1)
+      end
+    end
+    
+    def total_rooms
+      return @rooms
+    end
+    
+    def make_reservation(room_number, start_date, end_date, range)  
+      room = list_available_rooms(start_date, end_date).first
+      reservation = Hotel::Reservation.new(room, start_date, end_date, range)
+      return reservation
+    end
+    
+    def add_reservation(reservation)
+      @reservations << reservation
+    end 
+    
+    def list_reservations(date)
+      reservations_by_date = []
+      
+      @reservations.each do |reservation|
+        if reservation.date_range.include?(date)
+          reservations_by_date << reservation
+        end 
+      end
+      
+      if reservations_by_date.empty?
+        raise ArgumentError, "No reservations for this date."
+      end
+      #binding.pry - ok
+      return reservations_by_date
+    end
+    
+    def rooms_available?(room, start_date, end_date)            
+      reservations.each do |reservation|
+        
+        while start_date != end_date
+          if reservation.date_range.include?(start_date)
+            result = false
```

Comment:
You should return false as soon as you hit false, because you don't want to find an overlap and then reset to true, but you should not return true until you've looped through all the dates to make sure you haven't missed any overlaps.

Reply:
Fixed the loop and made new tests that are passing.